lib/pbench/test/unit/agent/test_tool_group.py

```diff
@@ -268,3 +270,52 @@ def mock_read_text(path):
                 assert tg.get_label(host) == labels[host]
             assert tg.get_tools("bad-host") == {}
             assert tg.get_label("bad-host") == ""
+
+
+def test_gen_tool_groups():
+    """Verify the tool group directory generator."""
+
+    class MockPath:
+        """A simple ToolGroup mock to generate host names from different
+        tool group directories.
+        """
+
+        _groups = set(("heavy", "light", "medium"))
+
+        def __init__(self, name: str):
+            self._name = name
+
+        def glob(self, prefix: str):
+            if self._name == "no-tool-group-dirs":
+                # The special name which for a directory which "does not
+                # have tool groups".
+                return
+            for n in self._groups:
+                # Return the 3 tools groups that are found with the prefix
+                # attached.
+                yield Path(f"{prefix[:-2]}-{n}")
+
+    class MockToolGroup:
+        """A simple no-op ToolGroup class which just stores the constructor
+        arguments.
+        """
+
+        TOOL_GROUP_PREFIX = "tests-vX"
+
+        def __init__(self, name: str, pbench_run: str):
+            self.name = name
+            self.pbench_run = pbench_run
+
+    with mock.patch("pbench.agent.tool_group.Path", MockPath), mock.patch(
+        "pbench.agent.tool_group.ToolGroup", MockToolGroup
+    ):
```

Comment:
If you add parens, Black will format this better:
```
    with (
        mock.patch("pbench.agent.tool_group.Path", MockPath),
        mock.patch("pbench.agent.tool_group.ToolGroup", MockToolGroup),
    ):
```
and I think it will work the same.

Reply:
I'll try it.